File: Source/Padutronics.Windows.Win32.Api.CWrapper/D2D1/ID2D1Geometry.cpp
#include "ID2D1Geometry.hpp"

auto ID2D1Geometry_CombineWithGeometry(ID2D1Geometry& _this, ID2D1Geometry* inputGeometry, D2D1_COMBINE_MODE combineMode, const D2D1_MATRIX_3X2_F* inputGeometryTransform, FLOAT flatteningTolerance, ID2D1SimplifiedGeometrySink* geometrySink) -> HRESULT
{
    return _this.CombineWithGeometry(inputGeometry, combineMode, inputGeometryTransform, flatteningTolerance, geometrySink);
}

auto ID2D1Geometry_GetBounds(ID2D1Geometry& _this, const D2D1_MATRIX_3X2_F* worldTransform, D2D1_RECT_F* bounds) -> HRESULT
{
    return _this.GetBounds(worldTransform, bounds);
}

File: Source/Padutronics.Windows.Win32.Api/D2D1/ID2D1Geometry.cs
using Padutronics.Windows.Win32.Api.DCommon;
using System.Runtime.InteropServices;

namespace Padutronics.Windows.Win32.Api.D2D1;

[Guid("2cd906a1-12e2-11dc-9fed-001143a055f9")]
public interface ID2D1Geometry : ID2D1Resource
{
    void CombineWithGeometry(ID2D1Geometry inputGeometry, D2D1_COMBINE_MODE combineMode, D2D_MATRIX_3X2_F? inputGeometryTransform, float flatteningTolerance, ID2D1SimplifiedGeometrySink geometrySink);
}